unix.cpp

```diff
@@ -2,15 +2,59 @@
 #include "settings.h"
 #include <QDir>
 #include <QStandardPaths>
+#include <QString>
 
 namespace Sys {
 QString archiveName(void)
 {
     return "linux64.zip";
 }
 
+void migrateHomePath(void)
+{
+    QString legacyHomePath = QDir::homePath() + "/.unvanquished";
+    QString xdgDataHome = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
+    QString xdgHomePath = xdgDataHome + "/unvanquished";
+
+    if (QDir(legacyHomePath).exists()) {
+        if (!QDir(xdgHomePath).exists()) {
+            if (!QDir(xdgDataHome).exists()) {
+                if (!QDir().mkdir(xdgDataHome)) {
+                    qFatal("Could not create XDG data directory %s", qPrintable(xdgDataHome));
```

Comment:
We probably shouldn't fatal out without giving instructions to users on how to fix their problem.

Reply:
If they can't create `~/.local/share` they probably can't run their desktop too 😛

Reply:
My point was offer a suggestion like "check your permissions" or "go to this wiki page to see how to migrate your homepath manually"

Reply:
This qFatal is not about creating `~/.local/share/unvanquished`, it's about creating `~/.local/share`, it's not about migrating the homepath. If people reach this qFatal, it means the system is so broken there is almost no way for them to have been able to run the updater at first and this problem is definitely not a migration issue. If his `${HOME}` is mounted read-only or if his `${HOME}` is so full he even can't create that directory or his `${HOME}` is under someone else's permission, the updater will even not be able to write it's own config file neither do anything useful…